File: vjezbe_2/zad_2.py
import matplotlib.pyplot as plt 
import numpy as np
from math import pi

def kosi_hitac(brzina,kut):
    polozaji_h = []
    polozaji_x = []
    vremena = []

    kut = kut * pi / 180
    brzina_y = np.sin(kut) * brzina
    brzina_x = np.cos(kut) * brzina
    vrijeme = (brzina_y*2) / 9.81
    domet = brzina_x * vrijeme
    t = 10/(1500)
    x = 0
    for i in range (1500):
        x = x + t
        vremena.append(x)
        polozaj_h = (brzina_y * x) - (9.81 * (x**2) / 2)
        if polozaj_h > 0 :
            polozaji_h.append(polozaj_h)
            polozaj_x =  brzina_x * x
            polozaji_x.append(polozaj_x)

        else:
            polozaji_x.append(domet)
            polozaji_h.append(0)
        
    x_cord = [polozaji_x]
    y_cord = [polozaji_h]
    plt.scatter(x_cord,y_cord,s=2)
    plt.xlabel('domet(m)')
    plt.ylabel('visina(m)')
    plt.show()

    y_cord = [polozaji_x]
    x_cord = [vremena]
    plt.scatter(x_cord,y_cord, s=2)
    plt.xlabel('vrijeme(s)')
    plt.ylabel('domet(m)')
    plt.show()

    y_cord = [polozaji_h]
    x_cord = [vremena]
    plt.scatter(x_cord,y_cord, s=2)
    plt.xlabel('vrijeme(s)')
    plt.ylabel('visina(m)')
    plt.show()

File: vjezbe_7/zad_2.cpp
#include <iostream>
#include <particle.h>
#include <math.h>
using namespace std;

int main(){
    Particle p1(15,60,0,20,0.01);
    cout << "Domet je: " << p1.range() << " m\n" ;
    cout << "Vrijeme je: " << p1.time() << " s\n" ;

    Particle p2(132,43,4,45,0.01);
    cout << "Domet je: " << p2.range() << " m\n" ;
    cout << "Vrijeme je: " << p2.time() << " s\n" ;

    Particle p3(4,4,4,4,0.01);
    cout << "Domet je: " << p3.range() << " m\n" ;
    cout << "Vrijeme je: " << p3.time() << " s\n" ;
    
    return 0;
}